src/components/Auth/SetPasswordComponent/SetPasswordComponent.tsx

```diff
@@ -1,33 +1,125 @@
 
-    const {authService} = useSecureContext();
+  const { password: confirmPassword, setPassword: setConfirmPassword } =
+    usePasswordValidation();
+  const submitPassword = async (): Promise<void> => {
+    try {
+      setLoading(true);
+      if (!token || typeof token !== "string") {
+        token = "";
+      }
+      await authService.updatePassword(token, newPassword);
+      navigation.navigate("Login");
+      setLoading(false);
+    } catch (error) {
+      setRequestErrorStatus(true);
+      setLoading(false);
+    }
+  };
 
-    const [password, setPassword] = useState("");
-
-    //TODO implement this view
-    return (
-        <BaseStyleComponent title="Set a new Password" subTitle="Please enter the email address used to sign in to your account.">
-            TODO!
-            {/* <TextInputComponent label="Email address" placeholder="Enter email address" type="emailAddress" value={username} onChangeText={(val) => setUsername(val)} />
-            <NblocksButton 
-                title="Get a reset link"
-                type="primary"
-                onPress={() => resetPassword()}
-                disabled={!username}
-            /> */}
+  let formIsValid = false;
+  if (newPasswordIsValid) {
+    formIsValid = samePasswordValidator(newPassword, confirmPassword);
+  }
+  const renderErrorMessages = wasTouched && !newPasswordIsValid;
+  return (
+    <Fragment>
+      {loading ? (
+        <ActivityIndicator color="#32B768" style={{ flex: 1 }} size="large" />
+      ) : (
+        <BaseStyleComponent
+          title="Set a new Password"
+          subTitle="Set your new password which you will be using for any future logins."
+        >
+          <TextInputComponent
+            label="New Password"
+            placeholder="Enter your new password."
+            type="password"
+            value={newPassword}
+            onChangeText={(text) => onNewPasswordTextChangeValidation(text)}
+            onPressIn={() => setWasTouched(true)}
+            onBlur={() => setWasTouched(true)}
+          />
+          {/* useEffect with clean up function should set the state of was touched */}
+          {renderErrorMessages && newPasswordFeedbackLog[0].error ? (
+            <ValidationMessageComponent feedbackLog={newPasswordFeedbackLog} />
+          ) : null}
+          <TextInputComponent
+            label="Confirm Password"
+            placeholder="Repeat your password"
+            type="password"
+            value={confirmPassword}
+            onChangeText={(text) => setConfirmPassword(text)}
+          />
+          <NblocksButton
+            title="Set New Password"
+            type={!requestErrorStatus ? "primary" : "danger"}
+            onPress={() => submitPassword()}
+            disabled={!formIsValid || requestErrorStatus}
+          />
+          {requestErrorStatus && (
+            <Text style={{ color: colors.dangerColor }}>
+              The link has expired, you'll need to generate a new one again.
```

Comment:
Example: User gets linked into app, tries setting password, doesn't work, this message is shown, there's no back button, user don't know what to do next.

Can we add  a button here together with error message that will take you to forgot-password screen here?
This is how it's done in the ionic plugin as you've seen.

Reply:
#Fix: I have implemented dynamic rendering of submit button. Such that, when link has expired it hides itself and displays a redirect button for generating new link.